process/block/shardblock.go

```diff
@@ -1062,7 +1062,7 @@ func (sp *shardProcessor) CommitBlock(
 		return err
 	}
 
-	err = sp.blockChain.SetCurrentBlockHeaderAndRootHash(header, committedRootHash)
+	err = sp.setCurrentBlockInfo(header, headerHash, committedRootHash)
```

Comment:
seems like there can be several errors on the commit after this point, where the CurrentBlock is already set.

can you check if this data (especially the one related to the setting of current block info is reverted on the defer sp.RevertCurrentBlock(headerHandler) ?)

Reply:
no, there is no handling at the moment on revert related to chain handler.
i think that can be handled into execution manager if needed